source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+        return getData( datum, info.date );
+        
+    } else if( message==M_SETDATA ) {   //! Caller is requesting to set data
+        H_THROW("TempDOECLIM: sendMessage not yet implemented for message=M_SETDATA.");
+        //TODO: call setData below
+        //TODO: change core so that parsing is routed through sendMessage
+        //TODO: make setData private
+
+        
+    } else {                        //! We don't handle any other messages
+        H_THROW( "Caller sent unknown message: "+message );
+    }
+    
+    return returnval;
+}
+
+//------------------------------------------------------------------------------
+// documentation is inherited
+void TempDOECLIMComponent::setData( const string& varName,
+                              const message_data& data ) throw ( h_exception )
+{
+    using namespace boost;
+    
+    H_LOG( logger, Logger::DEBUG ) << "Setting " << varName << "[" << data.date << "]=" << data.value_str << std::endl;
+    
+    try {
+        if( varName == D_ECS ) {
+            H_ASSERT( data.date == Core::undefinedIndex() , "date not allowed" );
+            S = unitval::parse_unitval( data.value_str, data.units_str, U_DEGC );
+        } else if( varName == D_DIFFUSIVITY ) {
+            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            diff = unitval::parse_unitval( data.value_str, data.units_str, U_CM2_S );
+	} else if( varName == D_AERO_SCALE ) {
+            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            alpha = unitval::parse_unitval( data.value_str, data.units_str, U_UNITLESS );
+        } else if( varName == D_TGAV_CONSTRAIN ) {
+            H_ASSERT( data.date != Core::undefinedIndex(), "date required" );
+            tgav_constrain.set( data.date, unitval::parse_unitval( data.value_str, data.units_str, U_DEGC ) );
+        } else {
+            H_THROW( "Unknown variable name while parsing " + getComponentName() + ": "
+                    + varName );
+        }
+    } catch( bad_lexical_cast& castException ) {
+        H_THROW( "Could not convert var: "+varName+", value: " + data.value_str + ", exception: "
+                +castException.what() );
+    } catch( h_exception& parseException ) {
+        H_RETHROW( parseException, "Could not parse var: "+varName );
+    }
+}
+
+//------------------------------------------------------------------------------
+// documentation is inherited
+// TO DO: should we put these in the ini file instead?
+void TempDOECLIMComponent::prepareToRun() throw ( h_exception ) {
+    
+    H_LOG( logger, Logger::DEBUG ) << "prepareToRun " << std::endl;
+    
+    if( tgav_constrain.size() ) {
+        Logger& glog = Logger::getGlobalLogger();
+        H_LOG( glog, Logger::WARNING ) << "Temperature will be overwritten by user-supplied values!" << std::endl;
+    }
+    
+    dt = 1;         // years per timestep (this is hard-coded into Hector
```

Comment:
If this is hard-coded in Hector, then we should be getting it from wherever it is hard coded at.  What if we someday change the hard coded value?  Then this component will be inconsistent, and it will likely go unnoticed until it causes problems.  Or worse, it will silently generate incorrect results.